llvm/lib/Target/Mips/MipsISelLowering.cpp

```diff
@@ -876,21 +876,60 @@ static SDValue performANDCombine(SDNode *N, SelectionDAG &DAG,
 static SDValue performORCombine(SDNode *N, SelectionDAG &DAG,
                                 TargetLowering::DAGCombinerInfo &DCI,
                                 const MipsSubtarget &Subtarget) {
-  // Pattern match INS.
-  //  $dst = or (and $src1 , mask0), (and (shl $src, pos), mask1),
-  //  where mask1 = (2**size - 1) << pos, mask0 = ~mask1
-  //  => ins $dst, $src, size, pos, $src1
   if (DCI.isBeforeLegalizeOps() || !Subtarget.hasExtractInsert())
     return SDValue();
 
   SDValue And0 = N->getOperand(0), And1 = N->getOperand(1);
   unsigned SMPos0, SMSize0, SMPos1, SMSize1;
   ConstantSDNode *CN, *CN1;
+  uint64_t Pos = 0;
+
+  if ((And0.getOpcode() == ISD::AND && And1.getOpcode() == ISD::SHL) ||
```

Comment:
And0/And1 here are so confused.  Maybe they need more descriptive names.

Reply:
Yes, I would update them to FirstOperand and SecondOperand.

Reply:
I guess `FirstOperation` and `SencondOperation` may be better, as both of them have 2 Operands, too.
It is so confusion.

Reply:
Because I referred to other combine-related functions(eg `performSHLCombine` and `performANDCombine`)in this file, they use these two variables `FirstOperand` and `FirstOperand`.